app/main.py

```diff
@@ -305,3 +437,85 @@ def download_gen3_metadata_file(program: str, project: str, uuid: str, format: s
                                      f"attachment;filename={filename}.csv"})
     except Exception as e:
         raise HTTPException(status_code=NOT_FOUND, detail=str(e))
+
+
+#
+# iRODS
+#
+
+
+def get_data_list(collect):
+    collect_list = []
+    for ele in collect:
+        collect_list.append({
+            "id": ele.id,
+            "name": ele.name,
+            "path": ele.path
+        })
+    return collect_list
+
+
+@app.get("/collection")
+async def get_irods_collections():
+    """
+    Return all collections from the root folder.
+    """
+    try:
+        collect = SESSION.collections.get(
+            f"{iRODSConfig.IRODS_ENDPOINT_URL}")
+    except Exception as e:
+        raise HTTPException(status_code=NOT_FOUND, detail=str(e))
+    folders = get_data_list(collect.subcollections)
+    files = get_data_list(collect.data_objects)
+    return {"folders": folders, "files": files}
+
+
+@app.post("/collection")
+async def get_irods_collections(item: CollectionItem):
+    """
+    Return all collections from the required folder.
+    """
+    try:
+        if item.path == None:
+            raise HTTPException(status_code=BAD_REQUEST,
+                                detail="Missing field in request body")
+
+        collect = SESSION.collections.get(item.path)
+        folders = get_data_list(collect.subcollections)
+        files = get_data_list(collect.data_objects)
+        return {"folders": folders, "files": files}
+    except Exception as e:
+        raise HTTPException(status_code=NOT_FOUND, detail=str(e))
+
+
+@app.get("/download/data/{suffix}")
+async def download_irods_data_file(suffix: str):
+    """
+    Return a specific download file from iRODS.
+
+    :param suffix: Required iRODS file path.
+    :return: A file with data.
+    """
+    url_suffix = suffix.replace("&", "/")
+    try:
+        file = SESSION.data_objects.get(
+            f"{iRODSConfig.IRODS_ENDPOINT_URL}/{url_suffix}")
+        with file.open("r") as f:
+            if suffix.endswith(".xlsx"):
+                download_path = os.path.join(
+                    os.path.expanduser('~'), "Downloads")
+                pd.read_excel(f).to_excel(f"{download_path}/{file.name}")
+                # content = pd.read_excel(f).to_csv()
+                return "Successfully download!"
```

Comment:
This requires a bit of works. I believe the content is downloaded on the server and not send to the user for xlsx files.
The file is downloaded at a temporary directory but never get cleared up.
It would be great to check if there is an API similar to the boto3.get_object method which returns a StreamingBody object in the body instead of download the file into the harddisk.